### ISSUE TITLE
Accessibility/Section 508 Compliance (Origin: bugzilla #411328)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#2400]
status RESOLVED severity *major* in component *general* for *---*
Reported in version *1.5.1* on platform *Other*
Assigned to: Dimitri van Heesch

Original attachment names and IDs:
- _[doxygen_508_report.html](http://ftp.stack.nl/pub/users/dimitri/bugzilla/attachments//bug411328_83185_doxygen_508_report.html)_ (ID 83185)
- _[ftvhelp.cpp.patch](http://ftp.stack.nl/pub/users/dimitri/bugzilla/attachments//bug411328_83236_ftvhelp.cpp.patch)_ (ID 83236)

On 2007-02-23 19:10:54 +0000, Deborah Lee Soltesz wrote:

> The HTML documentation generated by doxygen is not completely Section 508 (http://www.section508.gov/) compliant. Specifically noted is the missing noframes tag required frames tags, but other tags and attributes may also be missing.
> 
> As a government software/web developer, the web pages I publish are required to be compliant. The public domain software we develop here is distributed to the public as both binaries and source, and we've been using doxygen to provide APIs for our packages, both for our own use and our end users.
> 
> Our agency runs periodic accessibility compliance scans of our web sites using HiSoftware AccMonitor. Since our doxygen generated documentation is on the web (as well as in our distros), the files get scanned and unfortunately red flagged as non-compliant. It would be really wonderful if a future version of doxygen would feature HTML output that passes the accessibility scans. I haven't received a thorough scan of all the files generated, but a quick perusal of the source shows me most of the Section 508 points are covered and there's just a couple missing items (noframes alternative for frames pages).
> 
> Example (Frames)
> ----------------
> Doxy doc: http://isis.astrogeology.usgs.gov/Object/Programmer/
> 
> Scan report:
> 
> A. 508 Standards, Section 1194.22, (a) A text equivalent for every non-text element shall be provided (e.g., via "alt", "longdesc", or in element content).
> Rule: 1.1.5 - All FRAMESET elements will be checked for the NOFRAMES element.
> * Failure - Outermost FRAMESET Element at Line: 4, Column: 1
> 
> I. 508 Standards, Section 1194.22, (i) Frames shall be titled with text that facilitates frame identification and navigation.
> * Rule: 12.1.1 - All FRAME elements are required to contain the title attribute.
> * Failure - FRAME Element at Line: 5, Column: 3
> * Failure - FRAME Element at Line: 6, Column: 3

On 2007-02-23 19:15:55 +0000, Deborah Lee Soltesz wrote:

> Created attachment 83185
> Accessibility report for doxy HTML frames page
> 
> AccMonitor full accessibility report for doxy frames HTML page, including a link to the doxygen-created page for which the report was generated. While the web pages in question were generated with doxygen 1.4.7, doxygen 1.5.1 p1 was used for confirmation before this bug report was submitted.

On 2007-02-24 13:43:47 +0000, Kevin McBride wrote:

> Thanks for taking the time to report this bug.  I have analyzed the source code of doxygen, and have come to the conclusion that the output is indeed NOT compliant to USA Federal Statue section 508.  I have produced a patch for the frameset (needs internationalization); however, there is still a lot of work to be done.
> 
> I have added the gnome-love and helpwanted keywords to this bug report, and I have bumped up the priority and severity fields in hopes of getting Dimitri's attention onto this bug.
> 
> Please feel free to report any further bugs you may find.

On 2007-02-24 13:44:45 +0000, Kevin McBride wrote:

> Created attachment 83236
> Frameset Patch

On 2007-02-24 16:57:40 +0000, Deborah Lee Soltesz wrote:

> Thank you Kevin! Hopefully this will get me past the next agency scan.

On 2007-02-25 12:38:43 +0000, Dimitri van Heesch wrote:

> I'll include the patch that Kevin provided in the next CVS update. This should solve the noframes issue. If you encounter other compliancy issues please attach them to the bug report.
> 

On 2007-02-27 16:53:14 +0000, Deborah Lee Soltesz wrote:

> Thank you! I'll see if I can request a recursive scan on the locations where I have various doxygen docs residing or do one myself.

On 2007-04-05 20:10:22 +0000, Dimitri van Heesch wrote:

> This bug was marked "assigned" by me some time ago, which means it should be
> fixed in version 1.5.2 and is hereby marked as such. I would kindly request you
> to check if this version indeed fixes the problem and reopen the bug report
> should you still see the same problem.
